spring-boot-admin-samples/spring-boot-admin-sample-oauth2/spring-boot-admin-sample-oauth2-authorization/pom.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2014-2019 the original author or authors.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>spring-boot-admin-sample-oauth2-authorization</artifactId>
+    <name>Spring Boot Admin Sample OAuth2 - Authorization Server</name>
+    <description>Spring Boot Admin Sample OAuth2 - Authorization Server</description>
+    <parent>
+        <groupId>de.codecentric</groupId>
+        <artifactId>spring-boot-admin-sample-oauth2</artifactId>
+        <version>${revision}</version>
+        <relativePath>..</relativePath>
+    </parent>
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.security.oauth.boot</groupId>
+            <artifactId>spring-security-oauth2-autoconfigure</artifactId>
```

Comment:
Spring team prepared [demo application](https://github.com/jzheaux/messaging-app) with new OAuth2 approach some time ago. In that demo they made use of [Cloudfoundry UAA](https://github.com/cloudfoundry/uaa)

Reply:
I think it depends on how easy it will be to set up _authorization server_ using Spring Security 5.3. If it will be the matter of adding **@SpringBootApplication class + application.yml**, then probably we could stick to native solution.

At this moment I know nothing about 5.3, I will check it out later.

Reply:
Ok, taking into consideration the fact that 5.3 is not on the horizon, and you are not quite satisfied with `spring-security-oauth2-autoconfigure`, I decided to update my solution and make use of _Cloudfoundry UAA_.

I played a bit with it, it was quite easy to setup. It supports all major grant types that we may be interested in: `authorization_code`, `client_cridentials`, `password`, etc.

At this point, I have just migrated already existing solution with `password` grant type.
If you are ok with Cloudfoundry UAA, we can close this thread and agree on grant type in another thread.

Whatever type we pick, it should be easy to change it in [uaa.yml](https://github.com/elijah-pl/spring-boot-admin/blob/sample/oauth-2-spring-security-5/spring-boot-admin-samples/spring-boot-admin-sample-oauth2/spring-boot-admin-sample-oauth2-authorization/config/uaa.yml)

Reply:
I still think, this sample shouldn't involve cloud foundry. May be the [OpenId connect Playground may prove useful](https://openidconnect.net/)

Reply:
The spring security team also announced that they won't provide any authorization server. They point users to keycloak or other oidc services.

Reply:
In order to make this a stand-alone experience which is good for a sample I believe it is possible to implement the authorization server application based on good old `spring-security-oauth2` while the resource server and the SBA would benefit from first-class oAuth2 support in Spring5.

Reply:
> The spring security team also announced that they won't provide any authorization server. They point users to keycloak or other oidc services.

Could you please provide the source link? As I can see in their [blog post](https://spring.io/blog/2019/11/14/spring-security-oauth-2-0-roadmap-update#no-authorization-server-support) (updated section), it seems like they changed their mind.

Anyway, I agree that this example, probably, should not focus too much on authorization server configuration - it should be more about the **integration** of SBA with Spring Security OAuth2.

My motivation for using Cloudfoundry UAA was the fact that it did not require an account creation on any 3rd party service in case someone wanted to run the example.

Reply:
> Could you please provide the source link? As I can see in their blog post (updated section), it seems like they changed their mind.

I have no other information. But currently for me their decision is unchanged.

Reply:
> We are no longer planning on adding Authorization Server support to Spring Security.

https://github.com/spring-projects/spring-security/wiki/OAuth-2.0-Features-Matrix